### PR TITLE
MIMIR-26: Add caching for parent type iteration

### PR DESCRIPTION
**Pull request by itemodahl**
Add new file for parent utils

MIMIR-26

### CODE REVIEW

File: src/main/resources/lib/ssb/parent.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  const parentContent: Content<object, DefaultPageConfig> | null = getContent({
+    key: parentPathKey
+  })
+
+  if (parentContent) {
+    if (parentContent.page.config || parentContent.type === 'portal:site') {
+      return parentContent.page.config.pageType ? parentContent.page.config.pageType : 'default'
+    } else {
+      return fromParentTypeCache(parentPathKey,() => parentType(parentPathKey))
```
Comment (erikgus) [inline src/main/resources/lib/ssb/parent.ts:26]: One small lint error here

**Review by erikgus — APPROVED** [review]
Had one comment on lint error, other than that it looks good

File: src/main/resources/lib/ssb/parent.ts (hunk 1 above)
Comment (itemodahl) [reply src/main/resources/lib/ssb/parent.ts]: Linted

**Comment by itemodahl** [comment]
Thanks a bunch

**Pull request merged**